### PR TITLE
Changes to support Raspberry Pi

### PR DESCRIPTION
**Pull request by dschowta**
Here are the procedure followed to port to Raspberry PI:

		1. {In build machine machine}compile libnd4j
				git clone https://github.com/dschowta/libnd4j.git
			- For cross compilation use this link:http://stackoverflow.com/questions/19162072/installing-raspberry-pi-cross-compiler
			- Use the 4.9 version of gcc (raspberrypi/tools/arm-bcm2708/arm-rpi-4.9.3-linux-gnueabihf/bin/arm-linux-gnueabihf-g++)
		3.{In build machine}  build using "./buildnativeoperations.sh -o rp2"
		4. {In build machine} Install maven 3.3.9 (3.0.X does not work)
		5. {In build machine} followed instructions mentioned in https://deeplearning4j.org/buildinglocally :
				export LIBND4J_HOME=<pathTond4JNI>
				
				# build and install nd4j to maven locally (using the forked nd4j specifically changed for raspberry pi)
				git clone https://github.com/dschowta/nd4j.git
				cd nd4j
				mvn clean  install -DskipTests -Dplatform=linux-arm -Dmaven.javadoc.skip=true -P linux,arm -pl '!:nd4j-cuda-8.0,!:nd4j-cuda-8.0-platform,!:nd4j-tests'
		6. {In build machine } Copy following jars from the build machine :
			mkdir myjarFolder &&  \
			cp nd4j-backends/nd4j-backend-impls/nd4j-native/target/nd4j-native-<version>.jar\
			nd4j-backends/nd4j-backend-impls/nd4j-native/target/nd4j-native-<version>-linux-arm.jar \
			nd4j-backends/nd4j-backend-impls/nd4j-native-platform/target/nd4j-native-platform-<version>.jar\
			nd4j-backends/nd4j-api-parent/nd4j-api/target/nd4j-api-<version>.jar\
			nd4j-backends/nd4j-api-parent/nd4j-native-api/target/nd4j-native-api-<version>.jar \
			myjarFolder/
		7. {In build machine }install these jars in maven using :
			mvn install:install-file -Dfile=<path to myjarFolder>\nd4j-native-<version>.jar -DgroupId=org.nd4j -DartifactId=nd4j-native -Dversion=<version> -Dpackaging=jar -DgeneratePom=true
			mvn install:install-file -Dfile=<path to myjarFolder>\nd4j-native-<version>-linux-arm.jar -DgroupId=org.nd4j -DartifactId=nd4j-native -Dversion=<version> -Dpackaging=jar -DgeneratePom=true -Dclassifier=linux-arm
			mvn install:install-file -Dfile=<path to myjarFolder>\nd4j-api-<version>.jar -DgroupId=org.nd4j -DartifactId=nd4j-api -Dversion=<version> -Dpackaging=jar -DgeneratePom=true
			mvn install:install-file -Dfile=<path to myjarFolder>\nd4j-native-platform-<version>.jar -DgroupId=org.nd4j -DartifactId=nd4j-native-platform -Dversion=<version> -Dpackaging=jar -DgeneratePom=true
			mvn install:install-file -Dfile=<path to myjarFolder>\nd4j-native-api-<version>.jar -DgroupId=org.nd4j -DartifactId=nd4j-native-api -Dversion=<version> -Dpackaging=jar -DgeneratePom=true
			
		8. {In build machine } build the source of dependant appllication with above (step 7) dependencies.
		9. {In raspbian }copy the generated jar of dependant application to raspberry
		10. {In raspbian }download the libraries inside the folders of build machine nd4j-backends/nd4j-backend-impls/nd4j-native/target/classes/org/nd4j/nativeblas/linux-arm/  to a permanent folder (if possible to a system folder)
		11.{In raspbian }export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:<the path to libraryy>
		12. {In raspbian }java -jar myjar.jar

### CODE REVIEW

**Comment by saudet** [comment]
Thanks for the contribution! The same as with libnd4j, the instructions are bit more complicated than they need to be, but looks fine to me. And the same couple of things:
- The previous content of the README.md file is gone: Please restore the old README.md file and put your content into a new .md file (RaspberryPi2.md?)
- Instead of "rp2", could we use "linux-armhf" as the platform name to be consistent with the JavaCPP Presets? Accordingly please use "linux-armhf" instead of "linux-arm" for JavaCPP.

File: nd4j-backends/nd4j-backend-impls/nd4j-native-platform/pom.xml [hunk 1]
```diff
@@ -43,6 +44,12 @@
             <version>${project.version}</version>
             <classifier>${javacpp.platform.linux-ppc64le}</classifier>
         </dependency>
+		<dependency>
```
Comment (saudet) [inline nd4j-backends/nd4j-backend-impls/nd4j-native-platform/pom.xml]: Let's keep this in comment until we officially support this platform

File: pom.xml [hunk 2]
```diff
@@ -261,8 +270,7 @@
         <process-classes.skip> false</process-classes.skip>  <!-- To skip native compilation phase: -Dprocess-classes.skip=true    -->
         <javacpp.platform>${os.name}-${os.arch}</javacpp.platform> <!-- For Android: -Dplatform=android-arm                                        -->
         <javacpp.platform.root />            <!--              -Dplatform.root=/path/to/android-ndk/                         -->
-        <javacpp.platform.compiler />    <!--              -Dplatform.compiler=/path/to/arm-linux-androideabi-g++        -->
```
Comment (saudet) [inline pom.xml]: Not really important, but please fix the formatting while you're at it, such as here, but also in other places, replacing tab characters with an appropriate amount of spaces. Thanks!

Comment (dschowta) [inline pom.xml]: applied the changes

Comment (saudet) [inline pom.xml]: Hum, formatting is still broken here?

**Pull request merged**